Sunlay ggx field converter

SunlayGGX opened the pull request.

Commit:
gitignore

added the ignore field for ignoring visual studio generated file

Commit:
Beginning

I made the class Vertex as well as Vect3f.
- Vect3f is a simple class for storing coordinate
- Vertex is a class that stores Vertex informations (position and normal)
  and parse datas into std::string. Ready to save into a file.

Commit:
Rename

Rename my project from french name to english name...

Commit:
gitignore

Gitignore that ignore all files into Debug, Release and x64 doc for every
present and future folder

Commit:
Vertex and Vect3f classes

Tested and approuved!

- Vect3f implemented separetly from Vertex header file -> now independent
- Vect3f : a lot of operator overloading and method that principally do :
     -> add (AB + BC = AC)
     -> subtract (OA = A - O)
     -> scalar product
     -> cross product (vectorial product)
     -> multiplying by a scalar
     -> dividing by a scalar
- Some code commenting
- A unitary test proto inside a dirty method into main.cpp to check the
  good use of Vert3f

Commit:
Test heightmap sample

Here, I just added a raw uint8 file image generated by Terraingen.
It will be for testing and debugging purposes.

Commit:
The header of Heightmap class

Not functional

I've made a class (HeightMap) those purpose is to store and manipulate
directly the Heightmap loaded into an uint8_t array. No algorithm or image
processing here. Just some storing purposes.

Commit:
HeightMap.h

Not functional

- I added some getter really important (like size(), height(), ...).
- I commented my code.

Commit:
Some modification...

Commit:
Implementation HeightMap

Not tested!

-> Implementation all method

Commit:
Loading of the HeightMap File

Tested ! Functional but not approuved yet.

The HeightMap file is loading correctly into my buffer.

Though, some ameliorations could be done :
  - The path is litteral. We can change it to be dynamic.
  - The name of the file is defined by default. We could ask for the user
    to select a raw file he wants to load
  - The size of the file is defined by default too. We could change it

But for now, it is not really important. The class works and I'm happy.
Ameliorations can wait a little. The final result is more important.

SunlayGGX committed:
Merge pull request #1 from SunlayGGX/SunlayGGX-ReadHeightMap

Sunlay ggx read height map

obonnema commented on Projets/FieldConverter/FieldConverter/HeightMap.h:
```diff
@@ -0,0 +1,153 @@
+        enum
+        {
+            MIN_HEIGHT = 0,
+            MAX_HEIGHT = 255,
```
Deux Height différents, pas clair

obonnema reviewed (changes requested):
heightMap.h l 25

Commit:
Correction HeightMap constants name

Fix. Tested !
But not approuved yet for the same reason as the latter commit

Changed height map constants because there was a little misunderstanding
in the code.
Good now. Thank you Olivier.
